services/QuillLMS/app/services/duplicate_name_resolver.rb

```diff
@@ -4,9 +4,12 @@ class DuplicateNameResolver < ApplicationService
   attr_reader :name, :existing_names, :max_before_randomized
 
   MAX_BEFORE_RANDOMIZED = 10
+  MAX_STRING_LENGTH = 255
+  RANDOM_STRING_LENGTH = SecureRandom.urlsafe_base64.length
+  TRUNCATE_LENGTH = MAX_STRING_LENGTH - RANDOM_STRING_LENGTH
```

Comment:
We might also want to account for the `-` that's inserted by `name_with_random_string`.  So this might need another `- 1`?